internal/testsacc/edgegw_nat_rule_resource_test.go

```diff
@@ -1,285 +1,328 @@
+	return
 }
 
-func natRuleReflexiveTestCheck(resourceName string) resource.TestCheckFunc {
-	return resource.ComposeAggregateTestCheckFunc(
-		resource.TestCheckResourceAttrSet(resourceName, "id"),
-		resource.TestCheckResourceAttrSet(resourceName, "edge_gateway_id"),
-		resource.TestCheckResourceAttrSet(resourceName, "external_address"),
-		resource.TestCheckResourceAttr(resourceName, "name", "example-reflexive"),
-		resource.TestCheckResourceAttr(resourceName, "description", "description REFLEXIVE example"),
-		resource.TestCheckResourceAttr(resourceName, "internal_address", "192.168.0.1"),
-		resource.TestCheckResourceAttr(resourceName, "rule_type", "REFLEXIVE"),
-		resource.TestCheckResourceAttr(resourceName, "priority", "0"),
-		resource.TestCheckNoResourceAttr(resourceName, "snat_destination_address"),
-		resource.TestCheckNoResourceAttr(resourceName, "dnat_external_port"),
-	)
+func (r *NATRuleResource) Tests(ctx context.Context) map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test {
+	return map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test{
+		// * First test named "example"
+		"example": func(_ context.Context, resourceName string) testsacc.Test {
+			return testsacc.Test{
+				CommonChecks: []resource.TestCheckFunc{
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrWith(resourceName, "edge_gateway_id", uuid.TestIsType(uuid.Gateway)),
+					resource.TestCheckResourceAttr(resourceName, "name", "example"),
+					resource.TestCheckResourceAttrSet(resourceName, "description"),
+					resource.TestCheckResourceAttrSet(resourceName, "internal_address"),
+					resource.TestCheckResourceAttrSet(resourceName, "priority"),
+				},
+				// ! Create testing
+				Create: testsacc.TFConfig{
+					TFConfig: testsacc.GenerateFromTemplate(resourceName, `
+					resource "cloudavenue_edgegateway_nat_rule" "example" {
+						edge_gateway_id = cloudavenue_edgegateway.example.id
+
+						name        = "example"
+						rule_type   = "SNAT"
+						description = "description SNAT example"
+
+						# Using primary_ip from edge gateway
+						external_address         = "89.32.25.10"
+						internal_address         = "11.11.11.0/24"
+						snat_destination_address = "8.8.8.8"
+
+						priority = 10
+					}`),
+					Checks: []resource.TestCheckFunc{
+						// ! base checks
+						resource.TestCheckResourceAttr(resourceName, "external_address", "89.32.25.10"),
+						resource.TestCheckResourceAttr(resourceName, "description", "description SNAT example"),
+						resource.TestCheckResourceAttr(resourceName, "internal_address", "11.11.11.0/24"),
+						resource.TestCheckResourceAttr(resourceName, "rule_type", "SNAT"),
+						resource.TestCheckResourceAttr(resourceName, "snat_destination_address", "8.8.8.8"),
+						resource.TestCheckResourceAttr(resourceName, "priority", "10"),
+					},
+				},
```

Comment:
To avoid conflicts use template string generator. See example https://github.com/orange-cloudavenue/terraform-provider-cloudavenue/blob/main/internal/testsacc/vapp_isolated_network_resource_test.go#L50 

```suggestion
					TFConfig: testsacc.GenerateFromTemplate(resourceName, `
					resource "cloudavenue_edgegateway_nat_rule" "example" {
						edge_gateway_id = cloudavenue_edgegateway.example.id

						name        = {{ generate . "name" }}
						rule_type   = "SNAT"
						description = "description SNAT example"

						# Using primary_ip from edge gateway
						external_address         = "89.32.25.10"
						internal_address         = "11.11.11.0/24"
						snat_destination_address = "8.8.8.8"

						priority = 10
					}`),
					Checks: []resource.TestCheckFunc{
						// ! base checks
						resource.TestCheckResourceAttr(resourceName, "name", testsacc.GetValueFromTemplate(resourceName, "name")),					
						resource.TestCheckResourceAttr(resourceName, "external_address", "89.32.25.10"),
						resource.TestCheckResourceAttr(resourceName, "description", "description SNAT example"),
						resource.TestCheckResourceAttr(resourceName, "internal_address", "11.11.11.0/24"),
						resource.TestCheckResourceAttr(resourceName, "rule_type", "SNAT"),
						resource.TestCheckResourceAttr(resourceName, "snat_destination_address", "8.8.8.8"),
						resource.TestCheckResourceAttr(resourceName, "priority", "10"),
					},
				},
```